api/client/run.go

```diff
@@ -36,6 +38,23 @@ func (cid *cidFile) Write(id string) error {
 	return nil
 }
 
+// if container start fails with 'command not found' error, return 127
+// if container start fails with 'command cannot be invoked' error, return 126
+// return 125 for generic docker daemon failures
+func runReadBodyPostErr(err error) error {
+	statusError := Cli.StatusError{}
+	derr127 := errors.ErrorCodeCantStart127.Message()
+	derr126 := errors.ErrorCodeCantStart126.Message()
+	if derr127 == strings.Trim(err.Error(), "Error response from daemon: ") {
+		statusError = Cli.StatusError{StatusCode: 127}
+	} else if derr126 == strings.Trim(err.Error(), "Error response from daemon: ") {
+		statusError = Cli.StatusError{StatusCode: 126}
+	} else {
+		statusError = Cli.StatusError{StatusCode: 127}
+	}
+	return statusError
+}
+
```

Comment:
Not your fault, but I can't tell you how much this code makes me shutter.   strcmp (and even worse, substr) on error messages stinks.